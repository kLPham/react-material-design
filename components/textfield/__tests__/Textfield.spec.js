import React from 'react';
import { shallow } from 'enzyme';
import Textfield from '../index';

jest.mock('uuid', () => ({
    v4: jest.fn(() => '2e732f05-f466-4fba-acb5-2f3be58daf75'),
}));
describe('<Textfield />', () => {
    it('should render', () => {
        const textfield = shallow(
            <Textfield />,
    );
        expect(textfield).toMatchSnapshot();
    });
    it('should render with a label', () => {
        const label = 'this is my label';
        const textfield = shallow(
            <Textfield label={label} />,
    );
        expect(textfield).toMatchSnapshot();
    });
    it('should render with a value', () => {
        const value = 'this is my value';
        const label = 'this is my label';

        const textfield = shallow(
            <Textfield label={label} value={value} />,
    );
        expect(textfield).toMatchSnapshot();
    });
    it('should render with as primary', () => {
        const value = 'this is my value';
        const label = 'this is my label';
        const primary = true;
        const textfield = shallow(
            <Textfield label={label} value={value} primary={primary} />,
    );
        expect(textfield).toMatchSnapshot();
    });
    it('should render with a helpText', () => {
        const value = 'this is my value';
        const label = 'this is my label';
        const helpText = 'this is my helpText';
        const textfield = shallow(
            <Textfield label={label} value={value} helpText={helpText} />,
    );
        expect(textfield).toMatchSnapshot();
    });
    it('should render as required', () => {
        const label = 'this is my label';
        const required = true;
        const textfield = shallow(
            <Textfield label={label} required={required} />,
    );
        expect(textfield).toMatchSnapshot();
    });
});
