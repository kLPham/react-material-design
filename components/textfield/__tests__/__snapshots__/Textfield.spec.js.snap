// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Textfield /> should render 1`] = `
<FormField
  additionalClassNames="mdc-textfield"
  alignEnd={false}
>
  <input
    className="mdc-textfield__input"
    disabled={false}
    id="mdc-textfield--2e732f05-f466-4fba-acb5-2f3be58daf75"
    onChange={[Function]}
    required={false}
    type="text"
    value=""
  />
  <label
    className="mdc-textfield__label"
    htmlFor="mdc-textfield--2e732f05-f466-4fba-acb5-2f3be58daf75"
  />
</FormField>
`;

exports[`<Textfield /> should render as required 1`] = `
<FormField
  additionalClassNames="mdc-textfield"
  alignEnd={false}
>
  <input
    className="mdc-textfield__input"
    disabled={false}
    id="mdc-textfield--2e732f05-f466-4fba-acb5-2f3be58daf75"
    onChange={[Function]}
    required={true}
    type="text"
    value=""
  />
  <label
    className="mdc-textfield__label"
    htmlFor="mdc-textfield--2e732f05-f466-4fba-acb5-2f3be58daf75"
  >
    this is my label
  </label>
</FormField>
`;

exports[`<Textfield /> should render with a helpText 1`] = `
<FormField
  additionalClassNames="mdc-textfield"
  alignEnd={false}
>
  <input
    className="mdc-textfield__input"
    disabled={false}
    id="mdc-textfield--2e732f05-f466-4fba-acb5-2f3be58daf75"
    onChange={[Function]}
    required={false}
    type="text"
    value="this is my value"
  />
  <label
    className="mdc-textfield__label"
    htmlFor="mdc-textfield--2e732f05-f466-4fba-acb5-2f3be58daf75"
  >
    this is my label
  </label>
  <p
    className="mdc-textfield-helptext"
    id="mdc-textfield-helptext--2e732f05-f466-4fba-acb5-2f3be58daf75"
  >
    this is my helpText
  </p>
</FormField>
`;

exports[`<Textfield /> should render with a label 1`] = `
<FormField
  additionalClassNames="mdc-textfield"
  alignEnd={false}
>
  <input
    className="mdc-textfield__input"
    disabled={false}
    id="mdc-textfield--2e732f05-f466-4fba-acb5-2f3be58daf75"
    onChange={[Function]}
    required={false}
    type="text"
    value=""
  />
  <label
    className="mdc-textfield__label"
    htmlFor="mdc-textfield--2e732f05-f466-4fba-acb5-2f3be58daf75"
  >
    this is my label
  </label>
</FormField>
`;

exports[`<Textfield /> should render with a value 1`] = `
<FormField
  additionalClassNames="mdc-textfield"
  alignEnd={false}
>
  <input
    className="mdc-textfield__input"
    disabled={false}
    id="mdc-textfield--2e732f05-f466-4fba-acb5-2f3be58daf75"
    onChange={[Function]}
    required={false}
    type="text"
    value="this is my value"
  />
  <label
    className="mdc-textfield__label"
    htmlFor="mdc-textfield--2e732f05-f466-4fba-acb5-2f3be58daf75"
  >
    this is my label
  </label>
</FormField>
`;

exports[`<Textfield /> should render with as primary 1`] = `
<FormField
  additionalClassNames="mdc-textfield"
  alignEnd={false}
>
  <input
    className="mdc-textfield__input"
    disabled={false}
    id="mdc-textfield--2e732f05-f466-4fba-acb5-2f3be58daf75"
    onChange={[Function]}
    required={false}
    type="text"
    value="this is my value"
  />
  <label
    className="mdc-textfield__label mdc-theme--primary-bg mdc-theme--text-primary-on-primary"
    htmlFor="mdc-textfield--2e732f05-f466-4fba-acb5-2f3be58daf75"
  >
    this is my label
  </label>
</FormField>
`;
